pkg/urbit/vere/dawn.c

```diff
@@ -426,22 +426,44 @@ u3_dawn_vent(u3_noun seed)
     u3z(oct); u3z(fut);
   }
 
-  {
-    u3l_log("boot: retrieving sponsor keys\r\n");
+  pon = u3_nul;
+  while (c3__czar != rank) {
+    //  retrieve +point:azimuth of pos (sponsor of ship)
+    //
+    u3_noun who = u3dc("scot", 'p', u3k(pos));
+    c3_c* who_c = u3r_string(who);
+    u3z(who);
+    u3l_log("boot: retrieving keys for sponsor %s\r\n", who_c);
+    free(who_c);
```

Comment:
Style consideration: whenever it's possible, I'm a big fan of using blocks to keep "lifetimes" and variable scopes in sync for nouns. I think it makes it much easier to track reference counts. This is a minor case, but I'd still put it in a separate block.

Reply:
Done